datafusion/datasource/src/file_scan_config.rs

```diff
@@ -575,6 +575,95 @@ impl FileScanConfig {
         })
     }
 
+    /// Splits file groups into new groups based on statistics to enable efficient parallel processing.
+    ///
+    /// The method distributes files across a target number of partitions while ensuring
+    /// files within each partition maintain sort order based on their min/max statistics.
+    ///
+    /// The algorithm works by:
+    /// 1. Sorting all files by their minimum values
+    /// 2. Trying to place each file into an existing group where it can maintain sort order
+    /// 3. Creating new groups when necessary if a file cannot fit into existing groups
+    /// 4. Prioritizing smaller groups when multiple suitable groups exist (for load balancing)
+    ///
+    /// # Parameters
+    /// * `table_schema`: Schema containing information about the columns
+    /// * `file_groups`: The original file groups to split
+    /// * `sort_order`: The lexicographical ordering to maintain within each group
+    /// * `target_partitions`: The desired number of output partitions
+    ///
+    /// # Returns
+    /// A new set of file groups, where files within each group are non-overlapping with respect to
+    /// their min/max statistics and maintain the specified sort order.
+    pub fn split_groups_by_statistics_v2(
```

Comment:
The name is TBD.

I'm wondering if we need to keep the old method because I think both of them have applicable scenarios

Reply:
Keep the old and rename the new to `split_groups_by_statistics_with_target_partitions` or unify the two methods, both of them are okay for me. cc @alamb